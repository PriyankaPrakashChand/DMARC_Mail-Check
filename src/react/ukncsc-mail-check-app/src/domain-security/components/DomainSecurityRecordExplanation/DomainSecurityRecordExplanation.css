.DomainSecurityRecordExplanation--table {
  word-break: break-word;
  word-wrap: break-word;  
}
