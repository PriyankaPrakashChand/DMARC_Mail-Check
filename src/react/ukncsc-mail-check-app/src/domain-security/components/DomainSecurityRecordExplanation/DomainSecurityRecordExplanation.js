import React from 'react';
import { Header, Table } from 'semantic-ui-react';

import './DomainSecurityRecordExplanation.css';

const DomainSecurityRecordExplanation = ({ title, data }) =>
  !!data.length && (
    <React.Fragment>
      <Header as="h3">{title}</Header>
      <Table className="DomainSecurityRecordExplanation--table">
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell width={3}>Element</Table.HeaderCell>
            <Table.HeaderCell width={9}>Explanation</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          {data.map(({ value, explanation }) => (
            <Table.Row key={value}>
              <Table.Cell>{value}</Table.Cell>
              <Table.Cell>{explanation}</Table.Cell>
            </Table.Row>
          ))}
        </Table.Body>
      </Table>
    </React.Fragment>
  );

export default DomainSecurityRecordExplanation;
