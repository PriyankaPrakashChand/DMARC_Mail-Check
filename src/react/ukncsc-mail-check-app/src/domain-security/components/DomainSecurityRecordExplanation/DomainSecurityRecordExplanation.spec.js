import React from 'react';
import { render } from 'react-testing-library';
import DomainSecurityRecordExplanation from './DomainSecurityRecordExplanation';

describe('DomainSecurityRecordExplanation', () => {
  let container;

  beforeEach(() => {
    ({ container } = render(
      <DomainSecurityRecordExplanation
        title="Test"
        data={[{ value: 'foo', explanation: 'bar' }]}
      />
    ));
  });

  test('it should match the snapshot', () =>
    expect(container).toMatchSnapshot());
});
