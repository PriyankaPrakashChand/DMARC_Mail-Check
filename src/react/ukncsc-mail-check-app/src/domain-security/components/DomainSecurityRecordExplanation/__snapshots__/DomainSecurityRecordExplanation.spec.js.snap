// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecurityRecordExplanation it should match the snapshot 1`] = `
<div>
  <h3
    class="ui header"
  >
    Test
  </h3>
  <table
    class="ui table DomainSecurityRecordExplanation--table"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class="three wide"
        >
          Element
        </th>
        <th
          class="nine wide"
        >
          Explanation
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class=""
        >
          foo
        </td>
        <td
          class=""
        >
          bar
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
